DIG-1219: clinical_ETL_code as a package

daisieh opened the pull request:
If you download/pull the repo, you should be able to run:

```
# to install source code as a "package"
cd clinical_ETL_code
python -m pip install -e .
pytest
```

daisieh committed:
first draft of packaging

daisieh committed:
update code for module

daisieh committed:
remove deprecated tests

daisieh committed:
update test paths

daisieh committed:
add test for external mapping

daisieh committed:
install the code-module

daisieh committed:
Update pyproject.toml

mshadbolt commented:
It installs for me but I think we need some updates to the README file to show how to install and import as a package

daisieh committed:
allow any specified schema

daisieh committed:
add a schema for ingest's genomic samples

daisieh committed:
Update README.md

daisieh committed:
update dists

daisieh commented:
I added some info to the README, but then I added a bonus feature: you can now generally make a template for any openapi schema, if you make a schema class for it. I added one for the GenomicSample in ingest:

```
python src/clinical_etl/generate_schema.py --url https://raw.githubusercontent.com/CanDIG/candigv2-ingest/develop/ingest_openapi.yaml --schema GenomicSchema
```
will generate a template.csv based on that.

mshadbolt reviewed (approved):
Awesome, thanks for updating the info in the README. The new functionality to help with creating templates from generic openapi schemas and specifically the GenomicJSON seems like a really useful feature too!
